pulsar-broker/src/test/java/org/apache/pulsar/broker/stats/prometheus/NamespaceStatsAggregatorTest.java

```diff
@@ -0,0 +1,163 @@
+            multiLayerTopicsMap;
+
+    @BeforeMethod(alwaysRun = true)
+    public void setup() throws Exception {
+        multiLayerTopicsMap = ConcurrentOpenHashMap.<String,
+                        ConcurrentOpenHashMap<String, ConcurrentOpenHashMap<String, Topic>>>newBuilder()
+                .build();
+        pulsar = Mockito.mock(PulsarService.class);
+        broker = Mockito.mock(BrokerService.class);
+        doReturn(multiLayerTopicsMap).when(broker).getMultiLayerTopicMap();
+        Mockito.when(pulsar.getLocalMetadataStore()).thenReturn(Mockito.mock(ZKMetadataStore.class));
+        ServiceConfiguration mockConfig = Mockito.mock(ServiceConfiguration.class);
+        PulsarServiceMockSupport.mockPulsarServiceProps(pulsar, () -> {
+            doReturn(mockConfig).when(pulsar).getConfiguration();
+            doReturn(broker).when(pulsar).getBrokerService();
+        });
+    }
+
+    @Test
+    public void testGenerateSubscriptionsStats() {
+        // given
+        final String namespace = "tenant/cluster/ns";
+
+        // prepare multi-layer topic map
+        ConcurrentOpenHashMap bundlesMap = ConcurrentOpenHashMap.newBuilder().build();
+        ConcurrentOpenHashMap topicsMap = ConcurrentOpenHashMap.newBuilder().build();
+        ConcurrentOpenHashMap subscriptionsMaps = ConcurrentOpenHashMap.newBuilder().build();
+        bundlesMap.put("my-bundle", topicsMap);
+        multiLayerTopicsMap.put(namespace, bundlesMap);
+
+        // Prepare managed ledger
+        ManagedLedger ml = Mockito.mock(ManagedLedger.class);
+        ManagedLedgerMBeanImpl mlBeanStats = Mockito.mock(ManagedLedgerMBeanImpl.class);
+        StatsBuckets statsBuckets = new StatsBuckets(ManagedLedgerMBeanImpl.ENTRY_LATENCY_BUCKETS_USEC);
+        when(mlBeanStats.getInternalAddEntryLatencyBuckets()).thenReturn(statsBuckets);
+        when(mlBeanStats.getInternalLedgerAddEntryLatencyBuckets()).thenReturn(statsBuckets);
+        when(mlBeanStats.getInternalEntrySizeBuckets()).thenReturn(
+                new StatsBuckets(ManagedLedgerMBeanImpl.ENTRY_SIZE_BUCKETS_BYTES));
+        when(ml.getStats()).thenReturn(mlBeanStats);
+
+        // Prepare topic and subscription
+        PersistentTopic topic = Mockito.mock(PersistentTopic.class);
+        Subscription subscription = Mockito.mock(Subscription.class);
+        Consumer consumer = Mockito.mock(Consumer.class);
+        ConsumerStatsImpl consumerStats = new ConsumerStatsImpl();
+        when(consumer.getStats()).thenReturn(consumerStats);
+        when(subscription.getConsumers()).thenReturn(List.of(consumer));
+        subscriptionsMaps.put("my-subscription", subscription);
+        SubscriptionStatsImpl subStats = new SubscriptionStatsImpl();
+        TopicStatsImpl topicStats = new TopicStatsImpl();
+        topicStats.subscriptions.put("my-subscription", subStats);
+        when(topic.getStats(false, false, false)).thenReturn(topicStats);
```

Comment:
So whenever I add code to `NamespaceStatsAggregator` that calls to PersistentTopic (which will happen every time I have a new topic granularity metric), this test will fail. Mocking should be used cautiously, and when you mock way too much stuff, you should stop and think about what can be done differently. 

This test should have been deleted and added in `DispatcherBlockConsumerTest.testBlockDispatcherStats`

@aymkhalil @Technoboy- @codelipenghui